pkg/controller/dataimportcron-controller.go

```diff
@@ -264,24 +264,28 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 	desiredDigest := dataImportCron.Annotations[AnnSourceDesiredDigest]
 	digestUpdated := desiredDigest != "" && (len(imports) == 0 || desiredDigest != imports[0].Digest)
 	if digestUpdated {
+		updateDataImportCronFailingMetric(dataImportCron, true)
```

Comment:
why call it 4 times if you can call it once from updateDataImportCronCondition()?